torch/_inductor/ir.py

```diff
@@ -2917,45 +2917,41 @@ def create(cls, kernel, *args, **kwargs):
                 unflatten_args,
             ) = cls.process_kernel(kernel, *args, **kwargs)
 
-        if isinstance(example_output, (list, tuple)):
-            packed = FallbackKernel(
-                MultiOutputLayout(tensor_args[0].get_device()),
-                kernel,
-                tensor_args,
-                non_tensor_args,
-                unflatten_args,
-            )
-            return [
-                (
+        def generate_output(output, index=""):
+            if isinstance(output, list):
+                return list(
+                    generate_output(output[i], f"{index}[{i}]")
+                    for i in range(len(output))
+                )
+            elif isinstance(output, tuple):
+                return tuple(
+                    generate_output(output[i], f"{index}[{i}]")
+                    for i in range(len(output))
+                )
+            else:
+                return (
```

Comment:
should you assert on `output` type here, before trying to access its device and dtype?